helper/error-handler.ts

```diff
@@ -1,80 +1,82 @@
-      throw `[400] ${error.name ? `${error.name}. ` : ''}${error.message || 'Conflict'}`;
-    case 422:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Unprocessable Entity'}`;
-    case 429:
-      throw `[400] ${error.name ? `${error.name}. ` : ''}${error.message || 'Too Many Requests'}`;
-    case 500:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-    case 502:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Bad Gateway'}`;
-    case 503:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Service Unavailable'}`;
-    case 504:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Gateway Timeout'}`;
-    default:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-  }
-}
+import { ErrorStatusCode, HttpError } from '@errors/http/http-error';
+import { RuntimeError } from '@errors/runtime/runtime-error';
+import { log } from '@helper/logger';
+import { format } from '@redtea/format-axios-error';
+import { AxiosError } from 'axios';
+
+export interface AppError {
+  statusCode: number;
+  name: string;
+  message: string;
+}
+
+export function errorHandler(caughtError: Error | HttpError | AxiosError | RuntimeError): undefined {
+  let error = caughtError;
+  const axiosError = (caughtError as AxiosError).isAxiosError && format(caughtError as AxiosError);
+
+  if (!(error instanceof HttpError)) {
+    /**
+     * It means that error was unexpected and can have unpredictable structure
+     * For example, Axios errors have different structure
+     * Also, we can send all unexpected errors to Sentry here
+     */
+    log('This error was not generated by us. We should extract the statusCode, name and message here');
+
+    error = formatUnknownError(error, axiosError);
+  }
+
+  /**
+   * console.log allows displaying the error message and stack trace
+   * But Axios error has complicated structure that doesn't allow debugging it easily
+   */
+  if (axiosError) {
+    console.log(axiosError);
+  } else {
+    console.log(caughtError);
+  }
+
+  /**
+   * Serverless supports following status codes:
+   * 400  Bad Request
+   * 401  Unauthorized
+   * 403  Forbidden
+   * 404  Not Found
+   * 422  Unprocessable Entity
+   * 500  Internal Server Error
+   * 502  Bad Gateway
+   * 504  Gateway Timeout
+   */
+  error = formatUnsupportedError(error as HttpError | AppError);
+
+  /**
+   * The error message looks like: [404] Not Found. User does not exist
+   */
+  throw `[${(error as AppError).statusCode}] ${error.name}. ${error.message}`;
+}
+
+function formatUnknownError(error: Error | HttpError | AxiosError | RuntimeError, axiosError): AppError {
+  if (error instanceof RuntimeError) {
+    return { statusCode: 500, name: 'Bad request', message: error.message };

```

Comment:
What about converting InputValidationError to HttpBadRequestError?

Reply:
![image](https://user-images.githubusercontent.com/62546525/114829743-1f60d780-9dd4-11eb-9686-8c21126962b5.png)

Reply:
I meant adding one more convertion of InputValidationError to HttpBadRequestError

Reply:
`if (error instanceof InputValidationError) {
      return new HttpBadRequestError(error.message);
    }`
like this?

Reply:
Exactly